src/InputForm.tsx

```diff
@@ -0,0 +1,73 @@
+import { useState } from 'react';
+import { Todo } from './types/Todo';
+
+interface InputFormType {
+  setFilteredTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  setOnEmptyFormSubmit: React.Dispatch<React.SetStateAction<boolean>>,
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>
+}
+
+export const InputForm: React.FC<InputFormType>
+  = ({ setFilteredTodos, setOnEmptyFormSubmit, setTempTodo }) => {
+    const [inputValue, setInputValue] = useState('');
+    const [isInputDisabled, setIsInputDisabled] = useState(false);
+
+    const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setInputValue(event.target.value);
+    };
+
+    const handleFormSubmit = async (event:
+    React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      if (!inputValue) {
+        setOnEmptyFormSubmit(true);
+
+        return;
+      }
+
+      const todoData = {
+        id: 0,
+        userId: 10592,
```

Comment:
Consider fetching the userId from a global state or user context to make the code more reusable and not hard-coded.